src/compiler/utilities.ts

```diff
@@ -850,10 +850,88 @@ namespace ts {
         return node && (node.kind === SyntaxKind.GetAccessor || node.kind === SyntaxKind.SetAccessor);
     }
 
-    export function isClassLike(node: Node): node is ClassLikeDeclaration {
+    export function isClassLike(node: Node | undefined): node is ClassLikeDeclaration {
         return node && (node.kind === SyntaxKind.ClassDeclaration || node.kind === SyntaxKind.ClassExpression);
     }
 
+    export function isImmediatelyInvokedFunctionExpression(node: Node) {
+        if (!isCallExpression(node) ||
+            some(node.typeArguments) ||
+            some(node.arguments)) {
+            return false;
+        }
+        const expression = skipParentheses(node.expression);
+        if (!isFunctionExpression(expression) ||
+            some(expression.typeParameters) ||
+            some(expression.parameters) ||
+            expression.type ||
+            !expression.body) {
+            return false;
+        }
+        return true;
+    }
+
+    export function isTypeScriptClassWrapper(node: Node) {
+        if (!isImmediatelyInvokedFunctionExpression(node) ||
+            isParseTreeNode(node)) {
+            return false;
+        }
+
+        const func = <FunctionExpression>skipParentheses((<CallExpression>node).expression);

```

Comment:
`isImmediatelyInvokedFunctionExpression` already ensured the node has the correct shape.